src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/caseworker/ResetAndSendNewPinCallbackHandler.java

```diff
@@ -0,0 +1,137 @@
+import static uk.gov.hmcts.cmc.claimstore.services.ccd.Role.CASEWORKER;
+
+@Service
+public class ResetAndSendNewPinCallbackHandler extends CallbackHandler {
+
+    private static final List<Role> ROLES = Collections.singletonList(CASEWORKER);
+    private static final List<CaseEvent> EVENTS = ImmutableList.of(CaseEvent.RESET_PIN);
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    private CaseDetailsConverter caseDetailsConverter;
+
+    private UserService userService;
+
+    private CaseMapper caseMapper;
+
+    private ClaimIssuedNotificationService claimIssuedNotificationService;
+
+    private NotificationsProperties notificationsProperties;
+
+    private CCDCreateCaseService ccdCreateCaseService;
+
+    @Autowired
+    public ResetAndSendNewPinCallbackHandler(
+        CaseDetailsConverter caseDetailsConverter,
+        UserService userService,
+        CaseMapper caseMapper,
+        ClaimIssuedNotificationService claimIssuedNotificationService,
+        NotificationsProperties notificationsProperties,
+        CCDCreateCaseService ccdCreateCaseService
+    ) {
+        this.caseDetailsConverter = caseDetailsConverter;
+        this.userService = userService;
+        this.caseMapper = caseMapper;
+        this.claimIssuedNotificationService = claimIssuedNotificationService;
+        this.notificationsProperties = notificationsProperties;
+        this.ccdCreateCaseService = ccdCreateCaseService;
+    }
+
+    @Override
+    protected Map<CallbackType, Callback> callbacks() {
+        return ImmutableMap.of(
+            CallbackType.ABOUT_TO_SUBMIT, this::resendNewPin
+        );
+    }
+
+    @Override
+    public List<CaseEvent> handledEvents() {
+        return EVENTS;
+    }
+
+    @Override
+    public List<Role> getSupportedRoles() {
+        return ROLES;
+    }
+
+    private CallbackResponse resendNewPin(CallbackParams callbackParams) {
+        Claim claim = caseDetailsConverter.extractClaim(callbackParams.getRequest().getCaseDetails());
+        String authorisation = callbackParams.getParams().get(CallbackParams.Params.BEARER_TOKEN).toString();
+
+        if (claim.getDefendantId() != null) {
+            logger.info("Claim {} has already been linked to defendant - cannot send notification",
+                claim.getReferenceNumber());
+
+            return AboutToStartOrSubmitCallbackResponse.builder()
+                .errors(ImmutableList.of("Claim has already been linked to defendant - cannot send notification"))
+                .build();
+        }
+
+        if (!claim.getClaimData().getDefendant().getEmail().isPresent()) {
+            logger.info("Claim {} doesn't have defendant email address - cannot send notification",
+                claim.getReferenceNumber());
+
+            return AboutToStartOrSubmitCallbackResponse.builder()
+                .errors(ImmutableList.of("Claim doesn't have defendant email address - cannot send notification"))
+                .build();
+        }
```

Comment:
This check seems wrong. CCBC can still send new PINs to defendants by mail.

Reply:
If i understand you right, CCBC can probably send but resetPin event can't if there is no email address. There is another ticket coming up for case worked to be able to add new email from event.

Reply:
As discussed on Slack:

> Current functionality from the SupportController is only to email the staff the new PIN - they will do what they need with it, whether that’s to email it or mail it to the defendant

Please verify the requirements with Steve and update the A/C if necessary. As it stands, I expect the staff to trigger this, expect to receive an email themselves, and then possibly keep triggering it, flooding the defendant with emails.

Reply:
Steve Liddiard  8:27 PM
Hi - go live with what you have i.e. send link to defendant.  I'll ask Kirti to push forward with next part i.e. if defendant has no email we will send to support worker.